### ISSUE TITLE
Translate Pipe couldn't found

### ISSUE THREAD

**Issue by ZikZakJainy** [ngx-translate/core#459]
I am implementing Translation functionality in my application, and follow all the steps according to documentation but when, I implement translate pipe, it shows an error message mention below :

**Unhandled Promise rejection: Template parse errors:
The pipe 'translate' could not be found ("
    </form>

    <h4 class="text-muted text-thin nav-item">{{ [ERROR ->]'sidebar.WELCOME' | translate }} Language</h4>
    <select class="form-control" [ngModel]="selectedL"): Navbar@55:49 ; Zone: angular ; Task: Promise.then ;**

how to resolve this error ?

**Comment by SamVerschueren** [comment]
Make sure to import `TranslateModule` in the module of your component.

**Comment by ZikZakJainy** [comment]
Thanks for reply **SamVerschueren** , In my project different sub modules so, I imported **TranslateModule** in app module component like :
TranslateModule.forRoot({
            provide: TranslateLoader,
            useFactory: (createTranslateLoader),
            deps: [Http]
        }) 
but still translate pipe not fount.

**Comment by sazimi** [comment]
I also have the same issue!

**Comment by bc3** [comment]
i have the same issue, i also have a project with different sub modules and added the translatemodule to that specific component, no resolution ...

App.Modile also contains

  TranslateModule.forRoot({
            loader: {
                provide: TranslateLoader,
                useFactory: HttpLoaderFactory,
                deps: [Http]
            }
        })

when i remove code from template, it works, and when i try to access the translate service instance it gives me the correct info (chrome debug), i tried to add a translation to the service in code, no problem

i only have a problem when using the pipe in a template (not inline but with url)

**Comment by SamVerschueren** [comment]
It would be very helpful if one of you could create a plunkr to represent this problem.

Also make sure to import `TranslateModule` in your sub modules, **not only** in your app module.

**Comment by bc3** [comment]
@SamVerschueren , you are indeed correct, but i was adding the translate module into my component and not my sub module :-)

**Comment by jbgarr** [comment]
I was having this same issue as well. I have my main module (app.module.ts) and it handles the initial configuration. Then I have a shared module that is imported by all my other feature/page modules. In order for this to work for me I had to include the `TranslateModule` in my shared modules `imports` array as well as the `exports`. I would suggest adding this to the documentation because it's not obvious that the `TranslateModule` needs to be included in both arrays (unless I am doing something wrong here and this is not necessary).

**Comment by insanehong** [comment]
i have the same issue

**Comment by SamVerschueren** [comment]
@insanehong Did you read the answers? Are you 100% sure you added the `TranslateModule` to the `imports` array of your submodule as well? Recreate this with a plunkr and I will be happy to help.

**Comment by SamVerschueren** [comment]
> I would suggest adding this to the documentation because it's not obvious that the TranslateModule needs to be included in both arrays

Yes this! And it is documented right [here](https://github.com/ngx-translate/core#sharedmodule). This is not `ngx-translate` specific, you have to do it in for everything, that's just how the module system works. If you want to be able to use `*ngFor` in your templates for example, you have to import `CommonModule` every single time.
